Sources/CryptomatorCloudAccess/OneDrive/OneDriveAuthenticator.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  OneDriveAuthenticator.swift
+//  CryptomatorCloudAccess
+//
+//  Created by Philipp Schmid on 16.04.21.
+//  Copyright © 2021 Skymatic GmbH. All rights reserved.
+//
+
+#if canImport(CryptomatorCloudAccessCore)
+import CryptomatorCloudAccessCore
+#endif
+import MSAL
+import Promises
+import UIKit
+
+public enum OneDriveAuthenticatorError: Error {
+	case missingAccountIdentifier
+}
+
+public enum OneDriveAuthenticator {
+	public static func authenticate(from viewController: UIViewController) -> Promise<OneDriveCredential> {
+		let webviewParameters = MSALWebviewParameters(authPresentationViewController: viewController)
+		webviewParameters.webviewType = .safariViewController
```

Comment:
I would remove that line and use the default `webviewType`. I believe this was added due to our extensive tests. 😄